controllers/namespacelabel/namespacelabel_controller.go

```diff
@@ -97,112 +91,123 @@ var (
 	// - Keys are labels that might be set to the configMap object
 	// - Values are functions that receive the value of the label, the reconcile request,
 	//   and the reconciler instance.
-	configMapLabelBasedActions = map[string]func(string, reconcile.Request, *ReconcileNamespaceLabel) error{
+	configMapLabelBasedActions = map[string]func(string, reconcile.Request, *NamespaceLabelReconciler) error{
 		// Uninstall RHMI
 		"api.openshift.com/addon-rhmi-operator-delete": Uninstall,
 		// Uninstall MAO
 		"api.openshift.com/addon-managed-api-service-delete": Uninstall,
 	}
 )
 
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
+func (r *NamespaceLabelReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Namespace{}).
+		For(&corev1.ConfigMap{}).
+		Complete(r)
+}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	watchNS, err := k8sutil.GetWatchNamespace()
+func New(mgr manager.Manager) *NamespaceLabelReconciler {
+	watchNS, err := resources.GetWatchNamespace()
 	if err != nil {
-		cancel()
-		return nil, errors.Wrap(err, "could not get watch namespace from namespacelabel controller")
+		panic("could not get watch namespace from namespacelabel controller")
 	}
 	namespaceSegments := strings.Split(watchNS, "-")
 	namespacePrefix := strings.Join(namespaceSegments[0:2], "-") + "-"
 	operatorNs := namespacePrefix + "operator"
 
-	return &ReconcileNamespaceLabel{
-		mgr:               mgr,
-		client:            mgr.GetClient(),
-		scheme:            mgr.GetScheme(),
+	return &NamespaceLabelReconciler{
+		Client:            mgr.GetClient(),
```

Comment:
Why did these need to be converted to public attributes?

Reply:
In the case of `Client`, controllers generated by operator-sdk have now the client as an embedded field, so I folowed that pattern. I'll double check why I converted `Scheme`